client.go

```diff
@@ -175,3 +178,11 @@ func (c *Client) SetKeystoneToken(s string) {
 
 // ListOptions specifies the optional parameters for List method.
 type ListOptions struct{}
+
+func errorFromStatus(code int, msg string) error {
+	if code == http.StatusNotFound {
```

Comment:
Try add other error for status code that server can return.

Also use switch case instead of multi if/else if/else